packages/core/core/src/services/document-service/repository.ts

```diff
@@ -393,5 +393,15 @@ export const createContentTypeRepository: RepositoryFactoryMethod = (uid) => {
     publish: hasDraftAndPublish ? wrapInTransaction(publish) : (undefined as any),
     unpublish: hasDraftAndPublish ? wrapInTransaction(unpublish) : (undefined as any),
     discardDraft: hasDraftAndPublish ? wrapInTransaction(discardDraft) : (undefined as any),
+    /**
+     * @internal
+     * Exposed for use within document service middlewares
+     */
```

Comment:
you might want to add this internal tag in `types/src/modules/documents/service-instance.ts` so the tag is visible when users use the document service

Reply:
like this

```
  /**
   * @internal
   * Exposed for use within document service middlewares
   */
  updateComponents: (
    uid: Common.UID.Schema,
    entityToUpdate: {
      id: EntityService.Params.Attribute.ID;
    },
    data: EntityService.Params.Data.Input<Common.UID.Schema>
  ) => Promise<ComponentBody>;

  /**
   * @internal
   * Exposed for use within document service middlewares
   */
  omitComponentData: (
    contentType: Schema.ContentType | Schema.Component,
    data: EntityService.Params.Data.Input<Schema.ContentType['uid'] | Schema.Component['uid']>
  ) => Partial<
    EntityService.Params.Data.Input<Schema.ContentType['uid'] | Schema.Component['uid']>
  >;

```

and then just return them from here?

Reply:
yep! adding the internal tag in the types will display it everywhere